client/components/common/ProfilePictureModal.tsx

```diff
@@ -0,0 +1,141 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import { jsx } from '@emotion/core';
+import { Modal, notification } from 'antd';
+import { useTheme } from 'emotion-theming';
+import { useRouter } from 'next/router';
+
+import { useTranslation } from 'i18n';
+import { getImageUrl } from 'common/utils';
+import { PROFILE_PICTURES, mq } from '../../common/constants';
+import changeProfilePictureMutation from 'graphql/mutations/changeProfilePicture.graphql';
+import { Theme } from 'common/types';
+import {
+  changeProfilePicture,
+  changeProfilePictureVariables,
+} from 'graphql/mutations/__generated__/changeProfilePicture';
+import { useMutation } from '@apollo/client';
+
+interface Props {
+  visible: boolean;
+  onCancel: () => void;
+  currentlyActive: string;
+}
+
+const ProfilePictureModal: React.FC<Props> = ({
+  onCancel,
+  visible,
+  currentlyActive,
+}) => {
+  const { t } = useTranslation('common');
+  const [active, setActive] = useState<string>(currentlyActive);
+  const theme = useTheme<Theme>();
+
+  const [profilePictureMutate, { loading: changePictureLoading }] = useMutation<
+    changeProfilePicture,
+    changeProfilePictureVariables
+  >(changeProfilePictureMutation, {
+    variables: { picture: active },
+    awaitRefetchQueries: true,
+  });
+  const router = useRouter();
+
+  const onChangePicture = React.useCallback(
+    async (e: React.MouseEvent<HTMLElement>) => {
+      e.stopPropagation();
+      const { data } = await profilePictureMutate();
+      onCancel();
+      if (data?.changeProfilePicture?.ok) {
+        notification.success({
+          message: t('SUCCESS'),
+          // Missing translation
+          description: t('Profile picture changed successfully'),
+        });
+      }
+    },
+    [profilePictureMutate, router, onCancel],
+  );
+
+  const onCancelClick = React.useCallback(
+    (e: React.MouseEvent<HTMLElement>) => {
+      e.stopPropagation();
+      setActive(currentlyActive);
+      onCancel();
+    },
+    [onCancel],
+  );
+
+  return (
+    <Modal
+      visible={visible}
+      // Missing translation
+      title={'Change profile picture'}
+      centered
+      onOk={onChangePicture}
+      onCancel={onCancelClick}
+      confirmLoading={changePictureLoading}
+      okText={t('OK')}
+    >
+      <div
+        css={{ display: 'flex', justifyContent: 'center', marginBottom: 24 }}
+      >
+        <img
+          src={getImageUrl(active)}
+          // Missing translation
+          alt="Currently selected profile picture"
+          css={{ maxWidth: 220, borderRadius: 4 }}
+        />
+      </div>
+      <div
+        css={{
+          display: 'flex',
+          flexWrap: 'wrap',
+          alignItems: 'center',
+          justifyContent: 'center',
+          gap: '15px',
+          maxHeight: '30vh',
```

Comment:
Could we find a different way to do this? Maybe limit the modal as a whole? Seems a bit too "magic number" here

Reply:
Hmm, I guess it's doable, but my reasoning behind this wasn't finding a magic number, but for the selector to take a third of the screen. ∼1/3 for spacing on top and bottom, ∼1/3 for current selection, ∼1/3 for gallery. 30-33vh also ensures this ratio is maintained across all mobile screens (all of them have enough height for the component to not look weird too)